pages/index.js

```diff
@@ -91,7 +99,10 @@ export default function HomePage({ goals, categoryColors }) {
 
   useEffect(() => {
     const handleEscapeKeyPress = (e) => {
-      if (e.key === "Escape" && isModalOpen) {
+      if (
+        e.key === "Escape" &&
+        localStorage.getItem("isModalOpen") === "true"
```

Comment:
here you are still using the localStorage API directly

```suggestion
       isModalOpen
```